pkgs/tools/networking/tuic/default.nix

```diff
@@ -0,0 +1,41 @@
+{ lib
+, fetchFromGitHub
+, rustPlatform
+}:
+
+let
+  buildTUIC = name: rustPlatform.buildRustPackage rec{
+    pname = "tuic";
+    version = "1.0.0";
+
+    src = fetchFromGitHub {
+      owner = "EAimTY";
+      repo = pname;
+      rev = "${name}-1.0.0";
+      sha256 = "sha256-VoNr91vDqBlt9asT/dwCeYk13UNiDexNNiKwD5DSn8k=";
+    };
+
+    cargoPatches = [
+      ./0001-add-Cargo.lock.patch
+    ];
+
+    cargoHash = "sha256-EwrZzpSQ5cz1FAPZKTUS4DVCge1KE2Pna8XnPBWfcOE=";
+
+    cargoBuildFlags = "-p ${name}";
+
+    checkFlags = [ "--skip=lib" ];
+
+    meta = with lib; {
+      homepage = "https://github.com/EAimTY/tuic";
+      description = "Delicately-TUICed 0-RTT proxy protocol";
+      license = licenses.gpl3Only;
+      platforms = platforms.unix;
+      mainProgram = name;
+      maintainers = with maintainers; [ oluceps ];
+    };
+  };
+in
+{
+  server = buildTUIC "tuic-server";
+  client = buildTUIC "tuic-client";
```

Comment:
Maybe there is little benefit in splitting them into two packages?

Reply:
The tags seem to always point to the same commit, could be some limitations in there release tooling. I think the aesthetic issue is not a problem.